### PR TITLE
Begin accepting pull requests for gwtproject/gwt

### PR DESCRIPTION
**Pull request by niloc132**
Partial #9756

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (zbynek) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: maybe mention the possibility of using "fixes" keyword? https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests#linking-a-pull-request-to-an-issue
```suggestion
addressed in this pull request, e.g. "Fixes #42".
```

Comment (niloc132) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: Thoughts on the final line of the message I'm using for this commit, i.e. instead of `Fixes`, something like `Partial` to indicate that this commit is working to resolve an issue, but not itself enough to complete it?

Might be worth calling it out here, as plenty of issues are not one-and-done commits.

Something like this?

> addressed in this pull request, e.g. "Fixes #42" to close the linked issue, or "Partial #42" to indicate that this merge will address it, but that it isn't ready to be closed yet.

Comment (zbynek) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: Sounds good.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tbroyer) [inline .github/PULL_REQUEST_TEMPLATE.md:3]:
```suggestion
Please make sure your pull request has a short, concise title, and a clear
```

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (tbroyer) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: That said, the maintainer that merges the pull request will always have the opportunity to fix that in the squashed commit message.

What would happen if the pull request description says "Fixes #42" but this is actually only a partial fix, so the squashed commit is changed to "Partial #42"? I'm afraid it would mark the issue as fixed from the pull request description. IIRC, the maintainer should be able to edit the PR description to "Partial #42" before merging too.

So, well, the current text is OK for me, but the proposed one sounds good too :man_shrugging:

**Review by tbroyer — CHANGES_REQUESTED** [review]

**Comment by niloc132** [comment]
Irritatingly, I can't reply directly to this comment by @tbroyer 

> IIRC, the maintainer should be able to edit the PR description to Partial #42 before merging too.

Agreed - I propose that the description be generally used as a "draft commit message field", so it can be collaboratively edited there before the squash+merge.

(and as an aside, if anyone knows a way to get the "squash and merge" field to format your message so it correctly line wraps, I'd like to hear about it - I mostly copy out to an external text editor and adjust there)

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (niloc132) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: IIRC, the maintainer should be able to edit the PR description to Partial https://github.com/gwtproject/gwt/issues/42 before merging too.

Agreed - I propose that the description be generally used as a "draft commit message field", so it can be collaboratively edited there before the squash+merge.

(and as an aside, if anyone knows a way to get the "squash and merge" field to format your message so it correctly line wraps, I'd like to hear about it - I mostly copy out to an external text editor and adjust there)

Comment (jnehlmeier) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: Not sure if I like that "Partial" thing. 

Isn't the Github way to create a task list just like in https://github.com/gwtproject/gwt/issues/9756 and then convert each task into an issue? Then you fix the issues one-by-one using "fixed ..." and once all are done and their corresponding issues are closed via the pull requests the whole thing is done.

If you feel like one task needs two pull requests, then you would split the task into smaller tasks.

So in this concrete example here you make "Update pull request template with basic guidelines for developing on github first." an issue and then reference it using this pull requests.

Comment (niloc132) [reply .github/PULL_REQUEST_TEMPLATE.md:8]: I don't disagree that it wouldn't quite be the "github way", but I dont think that is necessarily compulsory either. I don't find "reopening issues after a bug is discovered" to be particularly the github way - the proper way seems like it ought to be to open a new issue rather than ever reopen anything. Plus, you run into the excitement of "well I merged to a release branch but not main (or vice versa), and the first sign of any merge github auto-closed the issue".

At least in other 10+ author projects I've been involved in, using a convention for "reference the ticket, but don't close it, and still look like other commit message trailers" has been helpful. I don't intend to drive any point through though, just offering this as a way to negotiate it.

Going the full other extreme would be actual `Bug-link: https://github.com/...` "trailers" on commit messages, but I do agree that it is nice to have "Fixes" message for cases where the bug is unambiguously fixed in a single commit.

**Comment by niloc132** [comment]
Updated as per discussion, if you have other feedback on this suggested path for pull request based development, let's hash it out soon to land this, or move to the mailing list for a follow-up.

**Review by tbroyer — APPROVED** [review]

**Review by zbynek — APPROVED** [review]

**Review by jnehlmeier — APPROVED** [review]

**Pull request merged**
